tests/atest/rules/naming/hyphen-in-variable-name/test.robot

```diff
@@ -0,0 +1,16 @@
+*** Test Cases ***
+My Test Case
+    ${r}    Set Variable  ${2-1}  # this is fine
+    ${a-b}  Set Variable  1       # this will warn - because if it's later used as ${a-b} it can lead to ambiguous results
+    ${-}    Set Variable  1       # this will warn - because if it's later used as ${a-b} it can lead to ambiguous results
+    ${a-}   Set Variable  1       # this will warn - because if it's later used as ${a-b} it can lead to ambiguous results
+    ${-b}   Set Variable  1       # this will warn - because if it's later used as ${a-b} it can lead to ambiguous results
+
+
+*** Keywords ***
+My Keyword
+    ${r}    Set Variable  ${2-1}  # this is fine
+    ${a-b}  Set Variable  1       # this will warn - because if it's later used as ${a-b} it can lead to ambiguous results
```

Comment:
What about ${a\-b}? (escaped char)

Reply:
I think it should also be reported because RF doesn't see that it is escaped (escaping `-` is not listed anywhere in RF documentation).